libs/sqlserver/langchain_sqlserver/vectorstores.py

```diff
@@ -225,31 +229,86 @@ def __init__(
         self._embedding_store = self._get_embedding_store(self.table_name, self.schema)
         self._create_table_if_not_exists()
 
-    def _can_connect_with_entra_id(self) -> bool:
-        """Determine if Entra ID authentication can be used.
+    def _get_connection_url(self, conn_string: str) -> str:
+        if conn_string is None or len(conn_string) == 0:
+            logging.error("Connection string value is None or empty.")
+            raise ValueError("Connection string value cannot be None.")
+
+        args = conn_string.split(";")
+        arg_dict = {}
+        for arg in args:
+            if "=" in arg:
+                # Split into key value pairs by the first positioned `=` found.
+                # Key-Value pairs are inserted into the dictionary.
+                #
+                key, value = arg.split("=", 1)
+                arg_dict[key.lower().strip()] = value.strip()
 
-        Check the components of the connection string to determine
-        if connection via Entra ID authentication is possible or not.
+        try:
+            # This will throw a key error if server or database keyword
+            # is not present in arg_dict from the connection string.
+            #
+            database = arg_dict.pop("database")
```

Comment:
Are the values "database" or "Database"? The example in the PR description seems to have, "Database" and here we are querying for "database". Is the String case insensitive?
`Driver={ODBC Driver 18 for SQL Server};Server=tcp:server.database.windows.net,1433;Database=dbname;Uid=username;Pwd=password;Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;`

Reply:
`key.lower().strip()` - This ensures the values are converted to lowercase values.